common/reviews/api/rush-lib.api.md

```diff
@@ -147,16 +147,22 @@ export class CredentialCache {
 }
 
 // @beta
-export type CustomTipId = 'TIP_RUSH_INCONSISTENT_VERSIONS' | string;
+export enum CustomTipIdEnum {
+    // (undocumented)
+    TIP_PNPM_NO_MATCHING_VERSION = "TIP_PNPM_NO_MATCHING_VERSION",
+    // (undocumented)
+    TIP_RUSH_INCONSISTENT_VERSIONS = "TIP_RUSH_INCONSISTENT_VERSIONS"
+}
 
 // @beta
 export class CustomTipsConfiguration {
     constructor(configFilename: string);
     readonly configuration: Readonly<ICustomTipsJson>;
-    showErrorTip(terminal: ITerminal, tipId: CustomTipId): void;
-    showInfoTip(terminal: ITerminal, tipId: CustomTipId): void;
-    showWarningTip(terminal: ITerminal, tipId: CustomTipId): void;
-    static readonly supportedTipIds: ReadonlySet<string>;
+    static CustomTipRegistry: Record<CustomTipIdEnum, ICustomTipInfo>;
+    showErrorTip(terminal: ITerminal, tipId: CustomTipIdEnum): void;
+    showInfoTip(terminal: ITerminal, tipId: CustomTipIdEnum): void;
+    showTip(terminal: ITerminal, tipId: CustomTipIdEnum): void;
+    showWarningTip(terminal: ITerminal, tipId: CustomTipIdEnum): void;
```

Comment:
These seem odd as members of the configuration class.

Reply:
Maybe we can make them `@internal` for now, and sort that out later?

```ts
/** @internal */
public _showTip(terminal: ITerminal, tipId: CustomTipIdEnum): void;
```

Reply:
I have made them `@internal`. 

> These seem odd as members of the configuration class.

True, do you think it is better to add them to the `Terminal`?  Then, anywhere with a `Terminal` instance could print tips by providing an ID.

Reply:
To the `Terminal` type in `node-core-library`? That probably doesn't make sense for other users of that type, but it could make sense to wrap that type in a class inside `rush-lib`.